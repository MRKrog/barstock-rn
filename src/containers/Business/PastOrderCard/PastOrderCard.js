import React, { Component } from "react";
import { connect } from "react-redux"
import { View, Text, ScrollView, TouchableOpacity, LayoutAnimation } from "react-native"
import styles from "./PastOrderCard.style"
import * as actions from "../../../redux/actions"

export class PastOrderCard extends Component{
  constructor(){
    super();
    this.state = {
      open: false
    }
  }

  toggle = () => {
    LayoutAnimation.configureNext(LayoutAnimation.create(100));
    this.setState({
      open: !this.state.open
    })
  }

  addToCart = () => {
    const { items } = this.props.order.attributes
    this.props.reOrder(items)
    this.props.navigation.navigate("Order")
  }

  render(){
    console.log("this is the alcohol info",this.props.alcohol)
    let orderItems = this.props.order.attributes.items.map(item => {
      let itemInfo = this.props.alcohol.find(alcohol => {
        return alcohol.id == item.id
      })
      console.log(itemInfo)
      return(
        <View style={styles.order_item}>
          <Text style={styles.order_name}>
            {itemInfo.attributes.name}
          </Text>
          <Text style={styles.order_quantity}>
            {item.quantity}
          </Text>
          <Text style={styles.order_price}>
            ${itemInfo.attributes.price * item.quantity} 
          </Text>
        </View>
      )
    })
    return(
      <View>
        <TouchableOpacity onPress={this.toggle} style={styles.order_btn}>
          <Text style={styles.order_title}>hello</Text>
        </TouchableOpacity>
        <View style={[{ height: this.state.open ? null : 0, overflow: 'hidden' }]}>
          <View style={styles.order_header}>
            <Text style={styles.order_rowOne}>Name</Text>
            <Text style={styles.order_rowTwo}>Quantity</Text>
            <Text style={styles.order_rowThree}>Price</Text>
          </View>
          <ScrollView>
              {
                orderItems
              }
          </ScrollView>
          <TouchableOpacity style={styles.order_reOrderBtn} onPress={this.addToCart}>
            <Text style={styles.order_reOrder}>Re Order</Text>
          </TouchableOpacity>
        </View>
      </View>
    )
  }
}

export const mapDispatchToProps = (dispatch) => ({
  reOrder: order => dispatch(actions.reOrder(order))
})

export const mapStateToProps = (state) => ({
  alcohol: state.alcohol
})

export default connect(mapStateToProps, mapDispatchToProps)(PastOrderCard)