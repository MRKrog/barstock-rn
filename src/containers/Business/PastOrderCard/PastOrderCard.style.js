import { Dimensions } from "react-native";

export default {
  order_btn: {
    backgroundColor: "#11212A",
    padding: 10,
    borderRadius: 5,
    marginTop: 10
  },
  order_reOrderBtn: {
    backgroundColor: "#EEAD0C",
    padding: 10,
    borderRadius: 5,
    marginTop: 10
  },
  order_title: {
    fontSize: 20,
    textTransform: "uppercase",
    fontFamily: 'abel',
    textAlign: "center",
    color: "#ffffff",
    letterSpacing: 1
  },
  order_reOrder: {
    fontSize: 15,
    textTransform: "uppercase",
    fontFamily: 'abel',
    textAlign: "center",
    color: "#ffffff",
    letterSpacing: 1
  },
  order_header: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    display: "flex",
    flexDirection: "row",
    paddingHorizontal: 15,
    paddingVertical: 5,
    backgroundColor: "#E8E8E8",
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderBottomColor: "#2D2D2D",
    shadowOpacity: 0.3,
    shadowRadius: 3,
    shadowColor: '#231f20',
    shadowOffset: { height: 1, width: 0 },
    zIndex: 9
  },
  order_rowOne: {
    flex: 1.5,
    fontSize: 13,
    fontFamily: 'Lato-Light',
    color: "#8e8e8e"
  },
  order_rowTwo: {
    flex: 1,
    fontSize: 13,
    fontFamily: 'Lato-Light',
    textAlign: "center",
    color: "#8e8e8e"
  },
  order_rowThree: {
    flex: 1,
    textAlign: "right",
    fontSize: 13,
    fontFamily: 'Lato-Light',
    color: "#8e8e8e"
  },
  order_item: {
    display: "flex",
    flexDirection: "row",
    justifyContent: "center",
    display: "flex",
    flexDirection: "row",
    paddingHorizontal: 15,
    paddingVertical: 5,
    paddingHorizontal: 15,
    paddingVertical: 10,
    borderBottomColor: "#2D2D2D",
    zIndex: 9
  },
  order_name: {
    flex: 1.5,
    fontSize: 13,
    fontFamily: 'Lato-Light',
    color: "#000000"
  },
  order_quantity: {
    flex: 1,
    fontSize: 13,
    fontFamily: 'Lato-Light',
    textAlign: "center",
    color: "#000000"
  },
  order_price: {
    flex: 1,
    textAlign: "right",
    fontSize: 13,
    fontFamily: 'Lato-Light',
    color: "#000000"
  },
  order_cardContainer: {
    marginLeft: 10,
    marginRight: 10,
  },
  order_itemsContainer: {
    backgroundColor: "#ffffff"
  }
};
