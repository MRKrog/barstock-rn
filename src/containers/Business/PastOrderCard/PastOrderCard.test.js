import { shallow } from "enzyme"
import { PastOrderCard } from "./PastOrderCard"
import React from "react"

describe("PastOrderCard", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(
      <PastOrderCard />
    )
  })

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot()
  })
})