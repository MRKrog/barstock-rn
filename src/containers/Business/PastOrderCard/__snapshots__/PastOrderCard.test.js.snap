// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastOrderCard should match snapshot 1`] = `undefined`;
